VARITY.pm

```diff
@@ -0,0 +1,184 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2023] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 CONTACT
+
+ Ensembl <http://www.ensembl.org/info/about/contact/index.html>
+
+=cut
+
+=head1 NAME
+
+ VARITY
+
+=head1 SYNOPSIS
+
+ mv VARITY.pm ~/.vep/Plugins
+ ./vep -i variations.vcf --assembly GRCh38 --plugin VARITY,file=/path/to/varity_all_predictions.txt
+
+ 
+=head1 DESCRIPTION
+
+ This is a plugin for the Ensembl Variant Effect Predictor (VEP) that
+ adds the VARITY score for missense variants to VEP output.
+
+ Please cite the VARITY publication alongside the VEP if you use this resource:
+ https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8715197/
+
+ Running options :
+
+ VARITY scores can be downloaded using wget http://varity.varianteffect.org/downloads/varity_all_predictions.tar.gz
+
+ The files can be tabix processed by :
+ tar -xzvf varity_all_predictions.tar.gz 
+ cat varity_all_predictions.txt | (head -n 1 && tail -n +2  | sort -t$'\t' -k 1,1 -k 2,2n) > varity_all_predictions_sorted.tsv
+ sed '1s/.*/#&/'  varity_all_predictions_sorted.tsv > varity_all_predictions.tsv  # to add a # in the first line of the file 
+ bgzip varity_all_predictions.tsv
+ tabix -f -s 1 -b 2 -e 2 varity_all_predictions.tsv.gz
+
+
+ Requirements:
+ The tabix utility must be installed in your path to use this plugin.
+ The --assembly flag is required to use this plugin.
+
+
+=cut 
+
+
+package VARITY;
+ 
+use strict;
+use warnings; 
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+use Bio::EnsEMBL::Variation::Utils::Sequence qw(get_matched_variant_alleles);
+
+sub new {
+  my $class = shift;
+  
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+  $self->get_user_params();
+  
+  my $params = $self->params_to_hash();
+
+  my $file;
+  if (!keys %$params) {
+    $file = $self->params->[0];
+    $params->{file} = $file;
+  } else {
+    $file = $params->{file};
+  } 
+  
+  $self->add_file($file);
+  
+  my $assembly = $self->{config}->{assembly};
+
+  if ($assembly ne "GRCh38") {
+    die "Assembly is not GRCh38, VARITY only works with GRCh38. \n";
+  }
```

Comment:
The plugin data only matches GRCh37 assembly. I don't think we should liftover the scores to GRCh38 given that the scores may not be appropriate then. Maybe we could ask the developers for a GRCh38 file in the future.

```suggestion
  if ($assembly ne "GRCh37") {
    die "Assembly is not GRCh37, VARITY only works with GRCh37. \n";
  }
```